### ISSUE TITLE
Accessibility: Autocomplete results are not announced when autocomplete is inside a dialog

### ISSUE THREAD

**Issue by 0ddfell0w** [angular/material#10804]
**Actual Behavior**:
 -  `What is the issue? *` When navigating an autocomplete list within an dialog, list values are not announced by screen reader.
 -  `What is the expected behavior?` As the user scrolls through items in the list, the values are read aloud.  

> “If an item is focused within the list, this should be announced next, followed by an indication of the item's position with the list if the screen reader supports this. As focus moves within the list, the screen reader should announce the relevant list items.” - [MDN Using the Listbox Role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_listbox_role)

**CodePen** (or steps to reproduce the issue): *
 -  `Steps to Reproduce`
    +  Activate your screen reader
    +  [Navigate to the Angular Material demo Page for mdAutocomplete](https://material.angularjs.org/latest/demo/autocomplete)
    + Open the example labelled "Inside Dialog"
    + Use your keyboard's directional keys to navigate through the items, notice that the values are not announced aloud
    + Repeat the process for any of the other autocomplete examples, notice the values *are* announced aloud.

 -  `Details:`

When a dialog is created that contains an autocomplete, the autocomplete first appends a live region [(div[role=status][aria-live=polite])](https://github.com/angular/material/blob/master/src/core/services/liveAnnouncer/live-announcer.js#L77) to the body of the document. [The dialog then hides everything in the DOM that is not an ancestor or child of the dialog](https://github.com/angular/material/blob/master/src/components/dialog/dialog.js#L1152), which unfortunately includes the new live region the new autocomplete just appended to the body. 

Thus, autocompletes within dialogs have live regions (`$mdLiveAnnouncer._liveElement`) with aria-hidden=true and that is why the autocomplete values are not read aloud.

**AngularJS Versions**: *
 -  `AngularJS Version:` 1.5.5
 -  `AngularJS Material Version:` 1.1.4

**Additional Information**:
 -  `Browser Type: *` Chrome
 -  `Browser Version: *` Version 59.0.3071.115 (Official Build) (64-bit)
 -  `OS: *` Mac OS X 10.12.5
 -  `Stack Traces:`

----
Shortcut to create a [new CodePen Demo](http://codepen.io/team/AngularMaterial/pen/bEGJdd).
Note: `*` indicates required information. Without this information, your issue may be auto-closed.

> Do not modify the titles or questions. Simply add your responses to the ends of the questions.
  Add more lines if needed.

@jelbourn
